src/ledger/LedgerStateCache.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+// Copyright 2025 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "ledger/LedgerTxn.h"
+#include "xdr/Stellar-ledger-entries.h"
+#include <optional>
+#include <shared_mutex>
+#include <unordered_set>
+
+namespace stellar
+{
+// TODO Add ifdef
+struct LedgerEntryHash
+{
+    size_t operator()(LedgerEntry const& entry) const;
+};
```

Comment:
This is a bit of a dangerous struct, since it's not obvious that this doesn't hash the whole key. I'd change the name to `LedgerEntryKeyHash` and maybe make it a public member of `LedgerStateCache`.